### PR TITLE
Add verbose logs

### PR DESCRIPTION
**Pull request by TwinProduction**
Truth be told, I'm looking to fix an issue that I have a feeling has to do with my usage of ASG's mixed instance policy, though I'm not even 100% sure.

In a nut shell, `aws-asg-roller` never stops updating instances, one at a time, as if it thinks that the template version is constantly updating. I also occasionally get the issue mentioned by @eterna2 on https://github.com/deitch/aws-asg-roller/pull/23#issuecomment-545232308 

Regardless, to debug this further, I need logs, hence this PR.

I initially wanted to pass a `verbose` parameter through every function, but I felt like it was needlessly intrusive.

I've also taken the opportunity to update the project to Go 1.13 and add the `vendor` folder to the project - hopefully you won't mind.

Let me know what you think.

---

**UPDATE**: Just remembered how obnoxiously big the vendor folder is whenever you have any AWS dependencies - so if the insanely obnoxious number of LoC bothers you, I can re-exclude the vendor folder.

---

**UPDATE**: I decided to re-remove the vendor folder after all, since it seems like it fails one of the CI's format check step.

### CODE REVIEW

**Comment by eterna2** [comment]
FYI

AWS just come up with managed node groups.

https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html

Still not everything I wanted, but closer now.

**Comment by TwinProduction** [comment]
@eterna2 Yeah, I'm aware, but I'm working on a non-managed work group spun up by Terraform that leverages Spot instances (with support for mixed instance type via `MixedInstancePolicy`), which isn't supported by managed worker groups.

**Comment by deitch** [comment]
Hey @TwinProduction (interesting handle), thanks for this. You are correct on both fronts:

* vendored libs can be quite large with AWS (but try kubernetes/client-go :-( )
* aws can be challenging to figure out what it is doing; it intentionally hides some of that "to make your life easier"

I am good with the verbosity, and with bumping to 1.13.x (although I know of people who push back on that, which is why we have git hash versions), and agree with you that we shouldn't vendor in the libs; `go.mod`/`go.sum` do it well enough.

Will review.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
-vendor/
 bin/
 .gocache/
 .go/
+.idea
```
Comment (deitch) [inline .gitignore]: Using intellij? No issues here. Can we split this into a separate PR?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
       uses: actions/checkout@v1
     - uses: actions/setup-go@v1
       with:
-        go-version: '1.12.10' # The Go version to download (if necessary) and use.
+        go-version: '1.13.7' # The Go version to download (if necessary) and use.
```
Comment (deitch) [inline .github/workflows/main.yml]: Also no issues here, but bumping go version is a big enough deal that I don't want to mix it with any other changes that aren't directly related to it (like verbosity). Can you split this into another PR?

File: aws.go [hunk 3]
```diff
@@ -76,6 +76,9 @@ func awsGetLaunchTemplate(svc ec2iface.EC2API, input *ec2.DescribeLaunchTemplate
 	return templatesOutput.LaunchTemplates[0], nil
 }
 func awsGetHostnames(svc ec2iface.EC2API, ids []string) ([]string, error) {
+	if len(ids) == 0 {
+		return []string{}, nil
+	}
```
Comment (deitch) [inline aws.go]: It obviously doesn't hurt, but why is it needed? Wouldn't the next lines get it? Or is it to avoid bothering making the API call in the first place?

File: roller.go [hunk 4]
```diff
@@ -26,12 +25,12 @@ func adjust(asgList []string, ec2Svc ec2iface.EC2API, asgSvc autoscalingiface.Au
 	// get information on all of the ec2 instances
 	instances := make([]*autoscaling.Instance, 0)
 	for _, asg := range asgs {
-		oldI, newI, err := groupInstances(asg, ec2Svc)
+		oldInstances, newInstances, err := groupInstances(asg, ec2Svc)
```
Comment (deitch) [inline roller.go]: You really don't like `oldI` and `newI`, do you? :-) 

That is fine. Sometimes my roots come out in terseness. I am good with something more explicit.

**Comment by deitch** [comment]
> AWS just come up with managed node groups

One step closer. Of course it is EKS-specific, but I expect eventually they will make all of ASG much better (or redundant).

File: aws.go (hunk 3 above)
Comment (TwinProduction) [reply aws.go]: You're right - it's to avoid making the API call in the first place.

The API call will in fact return a 400s because it expects at least one instance id to be passed as parameter, meaning that no matter what, the following line would never be reached:

```golang
if len(templatesOutput.LaunchTemplates) < 1 {
	return nil, nil
}
```

With this change, it will be reached, as intended.

File: .github/workflows/main.yml (hunk 2 above)
Comment (TwinProduction) [reply .github/workflows/main.yml]: Good point, reverted by a5170cb and 57b094f

File: .gitignore (hunk 1 above)
Comment (TwinProduction) [reply .gitignore]: Reverted by 3e0bfce

**Comment by TwinProduction** [comment]
Looks good to me, with these logs, I'll investigate the issue and probably create another PR to fix the issue itself.
